File: Commons/Geometry/src/test/java/com/greentree/commons/geometry/geom2d/AABBisIntersectTest.java
package com.greentree.commons.geometry.geom2d;

import com.greentree.commons.geometry.geom2d.shape.Circle2D;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

/**
 * @author Arseny Latyshev
 */
@DisplayName("AABB 2D isIntersect Tests")
public class AABBisIntersectTest {

    @DisplayName("Circle as AABB isIntersect")
    @Test
    public void Circle() {
        final var shape1 = new Circle2D(0, 0, 1);
        final var shape2 = new Circle2D(0, 0, 1);
        final var shape3 = new Circle2D(3, 0, 1);
        final var shape4 = new Circle2D(0, 3, 1);
        final var shape5 = new Circle2D(2, 2, 1.5f);
        assertTrue(shape1.getAABB().isIntersect(shape2.getAABB()));
        assertFalse(shape1.getAABB().isIntersect(shape3.getAABB()));
        assertFalse(shape1.getAABB().isIntersect(shape4.getAABB()));
        assertTrue(shape1.getAABB().isIntersect(shape5.getAABB()));
        assertFalse(shape2.getAABB().isIntersect(shape3.getAABB()));
        assertFalse(shape2.getAABB().isIntersect(shape4.getAABB()));
        assertTrue(shape2.getAABB().isIntersect(shape5.getAABB()));
        assertFalse(shape3.getAABB().isIntersect(shape4.getAABB()));
        assertTrue(shape3.getAABB().isIntersect(shape5.getAABB()));
        assertTrue(shape4.getAABB().isIntersect(shape5.getAABB()));

    }

}

File: Commons/Geometry/src/test/java/com/greentree/commons/geometry/geom2d/AABBisIntersectTest.kt
package com.greentree.commons.geometry.geom2d

import com.greentree.commons.geometry.geom2d.shape.Circle2D
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

/**
 * @author Arseny Latyshev
 */
@DisplayName("AABB 2D isIntersect Tests")
class AABBisIntersectTest {

	@DisplayName("Circle as AABB isIntersect")
	@Test
	fun Circle() {
		val shape1 = Circle2D(0f, 0f, 1f)
		val shape2 = Circle2D(0f, 0f, 1f)
		val shape3 = Circle2D(3f, 0f, 1f)
		val shape4 = Circle2D(0f, 3f, 1f)
		val shape5 = Circle2D(2f, 2f, 1.5f)
		assertTrue(shape1.boundingBox.isIntersect(shape2.boundingBox))
		assertFalse(shape1.boundingBox.isIntersect(shape3.boundingBox))
		assertFalse(shape1.boundingBox.isIntersect(shape4.boundingBox))
		assertTrue(shape1.boundingBox.isIntersect(shape5.boundingBox))
		assertFalse(shape2.boundingBox.isIntersect(shape3.boundingBox))
		assertFalse(shape2.boundingBox.isIntersect(shape4.boundingBox))
		assertTrue(shape2.boundingBox.isIntersect(shape5.boundingBox))
		assertFalse(shape3.boundingBox.isIntersect(shape4.boundingBox))
		assertTrue(shape3.boundingBox.isIntersect(shape5.boundingBox))
		assertTrue(shape4.boundingBox.isIntersect(shape5.boundingBox))
	}
}
